src/v/redpanda/admin/services/broker.cc

```diff
@@ -74,4 +85,106 @@ proto::admin::broker broker_service_impl::self_broker() const {
     return b;
 }
 
+namespace {
+
+struct connection_gather_result {
+    chunked_vector<proto::kafka_connection> connections;
+    size_t total_matching_count;
+};
+
+ss::future<connection_gather_result>
+gather_connections(const kafka::server& server, size_t limit) {
+    auto result = connection_gather_result{};
+
+    auto conn_ptrs = server.list_connections();
+    co_await ss::maybe_yield();
+
+    constexpr auto yield_period = 1000;
```

Comment:
travis has a helper in `ssx::async_for_each` that implements this, we should use it